### PR TITLE
Atom Tools: DocumentTypeInfo struct for multi document type support, code consolidation, comments, ~20 deleted images

### PR DESCRIPTION
**Pull request by gadams3**
TLDR multi document support, code consolidation, comments, ~20 deleted images 

•	DocumentTypeInfo describes a document type, a factory function for creating that document type, and mappings for different file types supported by operations like creating, opening, and saving a document. 
•	DocumentTypeInfo centralizes and standardizes a lot of the data that was needed for populating document creation dialogs, file selection dropdowns, and other UI. Because of that, a lot of code that previously differed between applications was able to be generalized and moved to a common location.
•	The asset browser interactions object is now contained in and managed by the atom tools application base class. All of the document related asset browser actions that were registered differently in each application have been generalized and moved to the atom tools document application base class.
•	The create document dialog was given a second constructor that takes in the document type info object pan fills everything else out automatically.
•	Instead of registering the factory function directly with the document system, we now register the document type info.
•	Updated the regular expression asset browser filter to support any regular expression and not just wildcards. This is posted in a separate PR but allowed the existing asset picker to be used as a general file picker with a source file filter supporting multiple selection and extensions.
•	The document type info changes also made it easy to extend the document system to support registering multiple document types in a single tool. In general, I would not recommend doing this for divergent or unrelated document types with different contexts, workflows, and UI requirements. However, I’m thinking it might facilitate sharing something like a lighting preset document between multiple tools. We would not want those automatically adding tabs to the main window or intermingling with the primary document types. Some minor changes need to be made anyway to support different or multiple views per document type instead of assuming all documents share the same type of view.
•	Miscellaneous clean up like deleting unused icons in different tools.

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetSelectionModel.cpp [hunk 1]
```diff
@@ -228,7 +228,7 @@ namespace AzToolsFramework
             return selection;
         }
 
-        AssetSelectionModel AssetSelectionModel::SourceAssetTypeSelection(const QString& pattern, bool multiselect)
+        AssetSelectionModel AssetSelectionModel::SourceAssetTypeSelection(const QRegExp& pattern, bool multiselect)
```
Comment (cgalvan) [inline Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetSelectionModel.cpp:231]: If the separate PR (https://github.com/o3de/o3de/pull/8059) for regular expression support instead of string/wildcard gets merged first, then you can pull that in so that it doesn't show up on the diff.

**Review by cgalvan — APPROVED** [review]

**Review by AMZN-mnaumov — APPROVED** [review]
